src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerUpdateApplicant1Email.java

```diff
@@ -0,0 +1,129 @@
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.divorce.caseworker.service.EmailUpdateService;
+import uk.gov.hmcts.divorce.common.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.divorcecase.model.Applicant;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+
+import static java.util.Collections.singletonList;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.POST_SUBMISSION_STATES;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE_DELETE;
+
+@Component
+@Slf4j
+public class CaseworkerUpdateApplicant1Email implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String CASEWORKER_UPDATE_APP1_EMAIL = "caseworker-update-app1-email";
+
+    @Autowired
+    private EmailUpdateService emailUpdateService;
+
+    private static final String EMAIL_LABEL = "${%s} email address";
+    private static final String APPLICANTS_OR_APPLICANT1S = "labelContentApplicantsOrApplicant1s";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_UPDATE_APP1_EMAIL)
+            .forStates(POST_SUBMISSION_STATES)
+            .name("Update applicant1 email")
+            .description("Update applicant/applicant1 email")
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .showSummary()
+            .showEventNotes()
+            .grant(CREATE_READ_UPDATE_DELETE, SUPER_USER, CASE_WORKER)
+            .grantHistoryOnly(
+                SOLICITOR,
+                LEGAL_ADVISOR))
+            .page("updateApp1Email")
+            .pageLabel("Update applicant/applicant1 email")
+            .complex(CaseData::getApplicant1)
+            .optionalWithLabel(Applicant::getEmail, getLabel(EMAIL_LABEL, APPLICANTS_OR_APPLICANT1S))
+            .done();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> midEvent(final CaseDetails<CaseData, State> details,
+                                                                  final CaseDetails<CaseData, State> detailsBefore) {
+        log.info("midEvent callback invoked for {}, Case Id: {}", CASEWORKER_UPDATE_APP1_EMAIL, details.getId());
+
+        CaseData caseData = details.getData();
+        CaseData caseDataBefore = detailsBefore.getData();
+
+        if (!validApplicant1ContactDetails(caseDataBefore, caseData)) {
```

Comment:
There is nothing on the ticket asking to restrict removal or email address. Since update contact details allowed caseworker to remove email in certain conditions (when represented or when offline) and I tried replicating that behaviour in these new events.

Based on your comment, I have asked a few additional questions of Fubara on the topic.

Reply:
Checking that user does not blank out email